frappe/modules/export_file.py

```diff
@@ -38,7 +39,35 @@ def write_document_file(doc, record_module=None, create_init=True):
 
 	# write the data file
 	fname = scrub(doc.name)
-	with open(os.path.join(folder, fname +".json"),'w+') as txtfile:
+	with open(os.path.join(folder, fname + ".json"), 'a+') as txtfile:
+		# if exporting DocType, retain order of 'fields' table and change order in 'field_order'
+		if doc.doctype == "DocType":
```

Comment:
move this to `doctype.py` and trigger `doc.trigger('before_export')`

never pollute general purpose functions with specific implementations. Beat's the whole purpose of encapsulation and simplicity.